README.md

```diff
@@ -39,35 +51,131 @@ class ListMovingVehicle(vqpy.QueryBase):
             )
 ```
 
-Under the hood, VQPy will automatically select an object detection model that outputs the specified `cls_name`. Multiple video optimizations will be conducted transparently to improve the end-to-end video query performance. 
+Under the hood, VQPy will automatically select an object detection model that outputs the specified `cls_name`, and decide the workflow to generate the object properties. Multiple video optimizations will be conducted transparently to improve the end-to-end video query performance.
 
-### Customization
+#### Detector and tracker
 
-For more details on customization, please refer to the document [here](https://github.com/uclasystem/VQPy/blob/main/docs/frontend.md#customization).
+Under normal conditions, users do not need to provide a detector implementation. However, if the user have a detector, they can inherit `vqpy.detector.DetectorBase` and implement the required interfaces.
+
+```python
+class DetectorBase(object):
+    """The base class of all predictors"""
+    cls_names = None        # the class names of the classification
+    output_fields = []      # the list of data fields the predictor can provide
+
+    def __init__(self, model_path: str) -> None:
+        self.model_path = model_path
+
+    def inference(self, img: np.ndarray) -> List[Dict]:
+        """Get the detected objects from the image
+        img (np.ndarray): the inferenced images
+        returns: list of objects, expressed in dictionaries
+        """
+        raise NotImplementedError
+```
+
+There are two level of trackers in VQPy. The surface level tracker stores the `VObj` instances by classes, and this is where `VObj` instances are created. For the default surface level tracker, different classes will be tracked separately using the ground level tracker. This level of tracker should generate the `track_id` field based on the provided detection result in this frame. Clearly, these trackers should be able to memorize historic detection results. To implement a method generating `track_id`, one should inherit `GroundTrackerBase` and register it to our backend. The interface of the ground level tracker is show below:
+
+```python
+class GroundTrackerBase(object):
+    """The ground level tracker base class.
+    Objects of this class approve detections results and associate the
+    results with necessary data fields.
+    """
+    input_fields = []       # the required data fields for this tracker
+    output_fields = []      # the data fields generated by this tracker
+    def __init__(self, stream: FrameStream):
+        ...
+    def update(self, data: List[Dict]) -> Tuple[List[Dict], List[Dict]]:
+        """Filter the detected data and associate output data
+        returns: the current tracked data and the current lost data
+        """
+        ...
+```
+
+For the registration details, please refer to our API document.
```

Comment:
We need to tell a whole story about how users should customize a tracker. Like first implemented your tracker inheriting tracker base and the supposed return value of `update`, then register it to model zoo with xxx.

Reply:
Same as customizing a detector (we need to demo how to register )

Reply:
Currently, there is no interface for registering a tracker. Registering a `GroundTracker` can be easy but `SurfaceTracker` is much more complex. Shall we add a simple interface for `GroundTracker` only?

Reply:
We can add a simple interface for `GroundTracker` only, or we can say that we don't support customizing `Tracker` for now (To support that, we need to change the internal tracker specification logic from `ByteTracker` to the user-customized one, and add checks to prevent error).

Reply:
I will resolve this after finishing the demo document.

Reply:
This interface should be added in the refactored code, and will be implemented in the upcoming PRs.